sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcColumnarBatchReaderSuite.scala

```diff
@@ -77,4 +77,44 @@ class OrcColumnarBatchReaderSuite extends QueryTest with SharedSparkSession {
       assert(p1.getUTF8String(0) === partitionValues.getUTF8String(0))
     }
   }
+
+  test("SPARK-32234: orc data created by the hive tables having _col fields name") {
+    withTable("test_date_hive_orc") {
+      spark.sql(
+        """
+          | CREATE TABLE test_date_hive_orc
+          | (_col1 INT, _col2 STRING, _col3 INT)
+          |  USING orc
+        """.stripMargin)
+      spark.sql(
+        """
+          | INSERT INTO
+          | test_date_hive_orc
+          | VALUES(9, '12', 2020)
+        """.stripMargin)
+
+      val df = spark.sql("SELECT _col2 from test_date_hive_orc")
+      checkAnswer(df, Row("12"))
+    }
+  }
+
+  test("SPARK-32234: orc data created by the spark having proper fields name") {
```

Comment:
Shall we remove this test case because this test case pass without your patch?

Reply:
done